src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -24,6 +24,8 @@ public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte
             {
                 case Oids.RsaRsa:
                     return BuildRsaPublicKey(encodedKeyValue);
+                case Oids.DsaDsa:
+                    return BuildDsaPublicKey(encodedKeyValue, encodedParameters);
```

Comment:
Just curious... the other two cases you modified earlier in the PR both have an EC variant, but this one has only RSA and DSA... I assume that's on purpose?